### PR TITLE
Code style enforcement, linting, package metadata

### PR DESCRIPTION
**Pull request by jvanstraten**
Noticed that the PyPI page for the current release is completely empty (not entirely sure why, but the sdist metadata is basically empty, aside from a curious link to LICENSE.txt, which was not shipped with the sdist; I didn't try to reproduce the problem, but building an sdist and wheel sets the metadata correctly for me now) and ended up copypasting most of the Python-related code quality stuff from the Substrait validator over to jdot.

 - Add formatting using black (CI + pre-commit)
 - Add linting using flake8 (CI + pre-commit)
 - Add license headers + check thereof (CI)
 - Add a bit more metadata to the package (trove classifiers etc)
 - Hopefully fix the complete lack of metadata on PyPI by explicitly including it and LICENSE.txt using MANIFEST.in

### CODE REVIEW

**Comment by anjakefala** [comment]
>  I didn't try to reproduce the problem, but building an sdist and wheel sets the metadata correctly for me now)

The setup.py was missing the relevant metadata! I noticed after having shipped to PyPi. Added it in for the next release.

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,2 @@
+include README.md
```
Comment (anjakefala) [inline MANIFEST.in:1]: Thanks for including the MANIFEST.in!

**Review by saulpw — APPROVED** [review]
Thanks @jvanstraten!  This is very welcome.

**Comment by saulpw** [comment]
But of course that this is now failing and delaying the merge is half of why I don't like black :)

**Comment by jvanstraten** [comment]
To be honest, I only use it because the VD Python guidelines tell me to, and because I do think some level of style enforcement is good in open source. I never use it in my own projects.

There's also a pre-commit hook for it that should be pinned to exactly the same version, so you don't need to rely on CI telling you it doesn't like a blank line somewhere. Should hopefully ease the pain a little :)

**Pull request merged**